ddtrace/profiling/collector/_lock.py

```diff
@@ -263,42 +263,36 @@ def _find_self_name(self, var_dict: typing.Dict):
         return None
 
     # Get lock acquire/release call location and variable name the lock is assigned to
+    # This function propagates ValueError if the frame depth is <= 3.
     def _maybe_update_self_name(self):
         if self._self_name is not None:
             return
-        try:
-            # We expect the call stack to be like this:
-            # 0: this
-            # 1: _acquire/_release
-            # 2: acquire/release (or __enter__/__exit__)
-            # 3: caller frame
-            if config.enable_asserts:
-                frame = sys._getframe(1)
-                if frame.f_code.co_name not in {"_acquire", "_release"}:
-                    raise AssertionError("Unexpected frame %s" % frame.f_code.co_name)
-                frame = sys._getframe(2)
-                if frame.f_code.co_name not in {
-                    "acquire",
-                    "release",
-                    "__enter__",
-                    "__exit__",
-                    "__aenter__",
-                    "__aexit__",
-                }:
-                    raise AssertionError("Unexpected frame %s" % frame.f_code.co_name)
-            frame = sys._getframe(3)
-
-            # First, look at the local variables of the caller frame, and then the global variables
-            self._self_name = self._find_self_name(frame.f_locals) or self._find_self_name(frame.f_globals)
-
-            if not self._self_name:
-                self._self_name = ""
-                LOG.debug(
-                    "Failed to get lock variable name, we only support local/global variables and their attributes."
-                )
-
-        except Exception as e:
-            LOG.warning("Error getting lock acquire/release call location and variable name: %s", e)
```

Comment:
Is this change meant to suppress the errors from, e.g. https://github.com/DataDog/dd-trace-py/issues/10963? There, the error log looks like `Error getting lock acquire/release call location and variable name: 'Model' object has no attribute '_has_params'`. It looks like, after this change, we would propagate that exception up from `_maybe_update_self_name` into the caller, `_acquire`, right? So it'd be caught & logged here instead: https://github.com/DataDog/dd-trace-py/blob/2d55a3c147e98a69fe9ec13f552594178c08f252/ddtrace/profiling/collector/_lock.py#L174 Is that any better? I'm assuming that, unless `config.enable_asserts` is enabled, then the exception would be coming from the `sys._getframe(3)` or `self._find_self_name` calls, right?

Reply:
I don't think the quality of information that we get is any better or worse. We're just handling it at a unified location. 

Yes, the exception would be coming from the `sys._getframe(3)`. I don't think `self._find_self_name` can raise an exception, correct me if I'm wrong.

Reply:
Okay, brain fart on my end. I see you changed the log level to debug. LGTM